lib/hanami/config/logger.rb

```diff
@@ -83,29 +83,35 @@ class Logger
       #   @since 2.0.0
       setting :filters, default: %w[_csrf password password_confirmation].freeze
 
+      # @!attribute [rw] logger_constructor
+      #   Sets or returns the constructor proc to use for the logger instantiation.
+      #
+      #   Defaults to `Dry.method(:Logger)`.
+      #
+      #   @api public
+      #   @since 2.0.0
+      setting :logger_constructor, default: Dry.method(:Logger)
+
       # @!attribute [rw] logger_class
       #   Sets or returns the class to use for the logger.
       #
-      #   This should be compatible with the arguments passed to the logger class' `.new` method in
-      #   {#instance}.
-      #
-      #   Defaults to `Hanami::Logger`.
```

Comment:
I agree. I left it there because I wanted to maintain the original functionality, but if  we're fine with removing it then I'm happy to do it in this PR.